R/vars-select.R

```diff
@@ -511,6 +517,26 @@ eval_minus <- function(expr, data_mask, context_mask) {
   -x
 }
 
+eval_or <- function(expr, data_mask, context_mask) {
+  x <- walk_non_symbol(expr[[2]], data_mask, context_mask)
+  y <- walk_non_symbol(expr[[3]], data_mask, context_mask)
+  c(x, y)
+}
+
+eval_and <- function(expr, data_mask, context_mask) {
+  x <- walk_non_symbol(expr[[2]], data_mask, context_mask)
+  y <- walk_non_symbol(expr[[3]], data_mask, context_mask)
+  set_intersect(x, y)
+}
+
+walk_non_symbol <- function(expr, data_mask, context_mask) {
+  if (is_symbol(expr)) {
```

Comment:
@hadley oh actually we should probably just fail with symbols instead of evaluating in the context. The error message should mention `all_of()` to refer to character vectors.

Reply:
Yeah, sounds good.

Reply:
```
> vars_select(letters, starts_with("a") & z)
Error: Can't use boolean operators with bare variables.
x `z` is a bare variable.
i Do you need `all_of(z)`

> vars_select(letters, starts_with("a") | z)
Error: Can't use boolean operators with bare variables.
x `z` is a bare variable.
i Do you need `all_of(z)`
```